crates/fuel-core/src/graphql_api/da_compression.rs

```diff
@@ -86,192 +86,395 @@ pub struct DecompressDbTx<'a, Tx, Onchain> {
+                    value: &$type,
+                    timestamp: Tai64,
+                ) -> anyhow::Result<()> {
+                    self.db_tx.write_registry(key, value, timestamp)
+                }
+
+                fn registry_index_lookup(
+                    &self,
+                    value: &$type,
+                ) -> anyhow::Result<Option<fuel_core_types::fuel_compression::RegistryKey>>
+                {
+                    self.db_tx.registry_index_lookup(value)
+                }
             }
 
-            fn write_registry(
-                &mut self,
-                key: &fuel_core_types::fuel_compression::RegistryKey,
-                value: &$type,
-                timestamp: Tai64,
-            ) -> anyhow::Result<()> {
-                self.db_tx.write_registry(key, value, timestamp)
+            impl<'a, Tx> EvictorDb<$type> for CompressDbTx<'a, Tx>
+            where
+                Tx: OffChainDatabaseTransaction,
+            {
+                fn set_latest_assigned_key(
+                    &mut self,
+                    key: fuel_core_types::fuel_compression::RegistryKey,
+                ) -> anyhow::Result<()> {
+                    self.db_tx.db_tx
+                        .storage_as_mut::<DaCompressionTemporalRegistryEvictorCache>()
+                        .insert(&MetadataKey::$type, &key)?;
+                    Ok(())
+                }
+
+                fn get_latest_assigned_key(
+                    &self,
+                ) -> anyhow::Result<Option<fuel_core_types::fuel_compression::RegistryKey>> {
+                    Ok(self
+                        .db_tx.db_tx
+                        .storage_as_ref::<DaCompressionTemporalRegistryEvictorCache>()
+                        .get(&MetadataKey::$type)?
+                        .map(|v| v.into_owned())
+                    )
+                }
             }
 
-            fn registry_index_lookup(
-                &self,
-                value: &$type,
-            ) -> anyhow::Result<Option<fuel_core_types::fuel_compression::RegistryKey>>
+        }};
+    }
+
+    impl_temporal_registry!(Address);
+    impl_temporal_registry!(AssetId);
+    impl_temporal_registry!(ContractId);
+    impl_temporal_registry!(ScriptCode);
+    impl_temporal_registry!(PredicateCode);
+};
+
+#[cfg(feature = "fault-proving")]
+const _: () = {
+    use v2::{
+        address::DaCompressionTemporalRegistryAddressV2,
+        asset_id::DaCompressionTemporalRegistryAssetIdV2,
+        contract_id::DaCompressionTemporalRegistryContractIdV2,
+        evictor_cache::DaCompressionTemporalRegistryEvictorCacheV2,
+        predicate_code::DaCompressionTemporalRegistryPredicateCodeV2,
+        registry_index::DaCompressionTemporalRegistryIndexV2,
+        script_code::DaCompressionTemporalRegistryScriptCodeV2,
+        timestamps::DaCompressionTemporalRegistryTimestampsV2,
+    };
+
+    macro_rules! impl_temporal_registry_v2 {
```

Comment:
This macro seems very repeated from the `impl_temporal_registry`. I think we should consider merging them. It seems like it's mostly the "V2" suffix which needs to be added in a bunch of places, and I believe we could just take that suffix as an argument to the macro. But that can be done as a follow-up, it doesn't have to block this PR.